### ISSUE TITLE
Can`t upload edited image

### ISSUE THREAD

**Issue by TekVanDo** [pqina/filepond#665]
I faced a problem  with filepond + doka 7+.
I turn off instantUpload
but if I will edit image in doka, button upload not apearing,
and I tryed no leasan some events like onconfirm or onpreparefile to trigger upload mannualy but it is not triggered 

my current config

```js
pondOptions = {
    labelIdle: 'Drag & Drop your picture or <span class="filepond--label-action">Browse</span>',
    // imageCropAspectRatio: '1:1',
    // stylePanelLayout: 'compact',
    styleButtonRemoveItemPosition: 'left bottom',
    styleButtonProcessItemPosition: 'right bottom',
    instantUpload: false,
    credits: null,
    server: {
      process: (fieldName, file, metadata, load, error, progress, abort) => {
        this.uploadFile(fieldName, file, metadata, load, error, progress, abort);
      },
      load: (uniqueFileId, load, error, progress, abort, headers) => {
        progress(true, 0, 1024);
        fetch(uniqueFileId)
          .then(res => {
            return res.blob();
          })
          .then(load);
      },
      fetch: null,
      revert: null
    },
    imagePreviewHeight: 256,
    filePosterMinHeight: 256,
    filePosterMaxHeight: 256,
    onpreparefile: (file, output) => {
      console.log(file, output, 'file')
      this.myPond.processFile(file).then(() => {
        console.log('done');
      });
    },
    onupdatefiles: (output) => {
      console.log('1aaaaaaaaaaaaaaaaaaaaa');
    },
    imageEditor: {
      // map legacy data objects to new imageState objects (only needs to be set when loading Doka v6 image data)
      legacyDataToImageState: legacyDataToImageState,
      // used to create the editor, receives editor configuration, should return an editor instance
      createEditor: openEditor,
      // Required, used for reading the image data
      imageReader: [createDefaultImageReader, { /* pass options here */ }],
      // optionally. can leave out when not generating a preview thumbnail and/or output image
      imageWriter: [createDefaultImageWriter, { /* pass options here */ }],
      // used to generate poster images, runs the editor without UI
      imageProcessor: processImage,
      // editor configuration options
      editorOptions: {
        onconfirm: (output) => {
          console.log('aaaaaaaaaaaaaaaaaaaaa');
        },
        onpreparefile: (file, output) => {
          console.log(file, output, 'file')
          // this.myPond.processFile(file).then(() => {
          //   console.log('done');
          // });
        },
        imageOrienter: imageOrienter,
        ...plugin_crop_defaults,
        ...plugin_finetune_defaults,
        ...plugin_filter_defaults,
        ...plugin_decorate_defaults,
        locale: {
          ...locale_en_gb,
          ...plugin_crop_locale_en_gb,
          ...plugin_finetune_locale_en_gb,
          ...plugin_filter_locale_en_gb,
          ...plugin_decorate_locale_en_gb,
        },
      },
    }
  }
```

**Comment by rikschennink** [comment]
I’ll respond to the GitHub issue.

Kind regards,

Rik | PQINA

Web: https://pqina.nl
Email: info@pqina.nl

PQINA BV
Sint Willibrordusstraat 16
5502 BH Veldhoven
The Netherlands

**Comment by rikschennink** [comment]
Can you check which version of FilePond you're on, I just noticed the included FilePond files in the preset template were very much outdated, best to download the latest versions from this repository. I'm updating the Doka Image Editor product package right now.

This method is only available on FilePond, you can't set it in the `editorOptions`
```
onpreparefile: (file, output) => {
```

Doka release v7.4.1 will enable binding `onconfirm` in the `editorOptions`, this is currently not supported but I can see how that might be useful.

This looks like it might cause this problem, `onpreparefile` will be called when the output image is generated, but then you instruct it to process the image again, which will trigger preparing of the file and in turn this callback.
```js
onpreparefile: (file, output) => {
      console.log(file, output, 'file')
      this.myPond.processFile(file).then(() => {
        console.log('done');
      });
},
```

**Comment by TekVanDo** [comment]
Ok i am waiting for a release. But it is strange that filepond don't react on doka image changes and don't suggest upload file.

**Comment by rikschennink** [comment]
@TekVanDo I'm not sure what you mean. 

Anyway, version 7.4.1 is now available for download, you can listen for Doka events using the new `handleEvent` property.

I'll close this issue, please submit a support ticket if you haven any further trouble.

**Comment by TekVanDo** [comment]
Hm it seems not working for me. handleEvent dont trigger any events

``
pondOptions = {
    labelIdle: `Drag & Drop your picture or <span class="filepond--label-action">Browse</span>`,
    // imageCropAspectRatio: '1:1',
    // stylePanelLayout: 'compact',
    styleButtonRemoveItemPosition: 'left bottom',
    styleButtonProcessItemPosition: 'right bottom',
    instantUpload: false,
    credits: null,
    server: {
      process: (fieldName, file, metadata, load, error, progress, abort) => {
        this.uploadFile(fieldName, file, metadata, load, error, progress, abort);
      },
      load: (uniqueFileId, load, error, progress, abort, headers) => {
        progress(true, 0, 1024);
        fetch(uniqueFileId)
          .then(res => {
            return res.blob();
          })
          .then(load);
      },
      fetch: null,
      revert: null
    },
    imagePreviewHeight: 256,
    filePosterMinHeight: 256,
    filePosterMaxHeight: 256,
    imageEditor: {
      // map legacy data objects to new imageState objects (only needs to be set when loading Doka v6 image data)
      legacyDataToImageState: legacyDataToImageState,
      // used to create the editor, receives editor configuration, should return an editor instance
      createEditor: openEditor,
      // Required, used for reading the image data
      imageReader: [createDefaultImageReader, {}],
      // optionally. can leave out when not generating a preview thumbnail and/or output image
      imageWriter: [createDefaultImageWriter, {}],
      // used to generate poster images, runs the editor without UI
      imageProcessor: processImage,
      // editor configuration options
      editorOptions: {
        imageOrienter: imageOrienter,
        ...plugin_crop_defaults,
        ...plugin_finetune_defaults,
        ...plugin_filter_defaults,
        ...plugin_decorate_defaults,
        locale: {
          ...locale_en_gb,
          ...plugin_crop_locale_en_gb,
          ...plugin_finetune_locale_en_gb,
          ...plugin_filter_locale_en_gb,
          ...plugin_decorate_locale_en_gb,
        },
        handleEvent: (type, detail) => {
          // log all editor events to console
          console.log(type, detail);
        }
      },
    }
  };
``

**Comment by rikschennink** [comment]
Fixed in 7.4.2
